utils/promtool.py

```diff
@@ -0,0 +1,27 @@
+import yaml
+import tempfile
+from subprocess import run, PIPE
+
+
+class PromtoolError(Exception):
+    pass
+
+
+def check_rule(yaml_spec):
+    try:
+        with tempfile.NamedTemporaryFile() as fp:
+            fp.write(yaml.dump(yaml_spec).encode())
+            fp.seek(0)
+            cmd = ['promtool', 'check', 'rules', fp.name]
+            status = run(cmd, stdout=PIPE, stderr=PIPE)
+    except Exception as e:
+        raise PromtoolError(f'Error building promtool file: {e}')
+
+    if status.returncode != 0:
+        message = 'Error running promtool'
+        if status.stderr:
+            message += ": " + status.stderr.decode()
+
+        raise PromtoolError(message)
+
+    return status.stdout
```

Comment:
This function returns either an string or an exception. Later, you get the exception message and build a dictionary that is returned from the threaded execution... it's a lot of transformations as the data you need goes by.
Can we simplify the transformations? This function could return a dict:
```
{
    valid: True | false
    message: status.stdout
}
```
Or any custom object that carries status + message, being the same object for both good and bad statuses. Then dict (or the custom object) would be directly processed by the comprehension getting the threaded results. Wdyt?

Reply:
It will simplify things. I will go that route.

Reply:
fixed in eb8e705